XIPs/xip-10-brandinfo-schema.md

```diff
@@ -0,0 +1,81 @@
+---
+xip: 14
+title: Brand info schema in conversation metadata
+description: Schema to enable standardized sharing of brand information as it relates to conversationId
+author: Yash Lunagaria (@yash-luna)
+status: Review
+type: Standards Track
+category: XRC
+created: 2023-01-31
+---
+
+## Abstract
+
+This XIP proposes a schema for sharing brand information as it relates to `conversationId`. A standard way to represent this information in conversation `metadata` will enable different front-ends to surface the context captured by `conversationId` in a consistent, user-friendly manner while still affording clients design and UI flexibility.
+
+## Motivation
+
+The SDK API currently accepts a unique `conversationId` per address pair and allows `metadata` to be set as well. Today, different apps use these parameters in different ways and therefore other apps may not know how to render the `conversationId` appropriately within the UI to reflect the corresponding brand. We propose a schema to standardize the use of conversation `metadata` for the purpose of client-side brand expression stemming from the `conversationId`. The `metadata` field may include other non-standard (or future standard) properties.
+
+Note that the schema is only applicable when setting a non-null `conversationId`. If `conversationId` is null, the SDK does not allow setting `metadata`.
+
+## Specification
+
+Proposed `brandInfo` schema in conversation `metadata`
+
+```json
+{
+    conversationId: "mydomain.xyz/abc/qrs",
+    metadata: { "brandInfo.displayName": "My company", "brandInfo.profileImage": "mydomain.xyz/assets/myimage.png", "brandInfo.primaryColor": "#ffffff" }
+}
+```
```

Comment:
What do you think of formatting it like this to make the metadata schema a little bit more readable? It's so good I want people to immediately see the value!

```
{
   "conversationId":"mydomain.xyz/abc/qrs",
   "metadata":{
      "brandInfo.displayName":"My company",
      "brandInfo.profileImage":"mydomain.xyz/assets/myimage.png",
      "brandInfo.primaryColor":"#ffffff"
   }
}
```